packages/repo-tools/src/commands/api-reports/api-extractor.ts

```diff
@@ -374,209 +377,237 @@ export async function runApiExtraction({
+        if (shouldLogInstructions) {
+          logApiReportInstructions();
 
-        if (conflictingFile) {
-          console.log('');
-          console.log(
-            `The conflicting file is ${relativePath(
-              tmpDir,
-              conflictingFile,
-            )}, with the following content:`,
-          );
-          console.log('');
+          if (conflictingFile) {
+            console.log('');
+            console.log(
+              `The conflicting file is ${relativePath(
+                tmpDir,
+                conflictingFile,
+              )}, with the following content:`,
+            );
+            console.log('');
 
-          const content = await fs.readFile(conflictingFile, 'utf8');
-          console.log(content);
+            const content = await fs.readFile(conflictingFile, 'utf8');
+            console.log(content);
 
-          logApiReportInstructions();
+            logApiReportInstructions();
+          }
         }
-      }
 
-      throw new Error(
-        `API Extractor completed with ${extractorResult.errorCount} errors` +
-          ` and ${extractorResult.warningCount} warnings`,
-      );
-    }
+        throw new Error(
+          `API Extractor completed with ${extractorResult.errorCount} errors` +
+            ` and ${extractorResult.warningCount} warnings`,
+        );
+      }
 
-    const warningCountAfter = await countApiReportWarnings(reportPath);
-    if (noBail) {
-      console.log(`Skipping warnings check for ${packageDir}`);
-    }
-    if (warningCountAfter > 0 && !noBail) {
-      throw new Error(
-        `The API Report for ${packageDir} is not allowed to have warnings`,
-      );
-    }
-    if (warningCountAfter === 0 && allowWarningPkg.includes(packageDir)) {
-      console.log(
-        `No need to allow warnings for ${packageDir}, it does not have any`,
-      );
-    }
-    if (warningCountAfter > warningCountBefore) {
-      warnings.push(
-        `The API Report for ${packageDir} introduces new warnings. ` +
-          'Please fix these warnings in order to keep the API Reports tidy.',
-      );
+      const warningCountAfter = await countApiReportWarnings(reportPath);
+      if (noBail) {
+        console.log(`Skipping warnings check for ${packageDir}`);
+      }
+      if (warningCountAfter > 0 && !noBail) {
+        throw new Error(
+          `The API Report for ${packageDir} is not allowed to have warnings`,
+        );
+      }
+      if (warningCountAfter === 0 && allowWarningPkg.includes(packageDir)) {
+        console.log(
+          `No need to allow warnings for ${packageDir}, it does not have any`,
+        );
+      }
+      if (warningCountAfter > warningCountBefore) {
+        warnings.push(
+          `The API Report for ${packageDir} introduces new warnings. ` +
+            'Please fix these warnings in order to keep the API Reports tidy.',
+        );
+      }
+      if (staleReportFiles.length) {
+        if (noBail) {
```

Comment:
```suggestion
        if (isLocalBuild) {
```

Thinking this is better? Just deletes the file when running locally, but errors out in CI.

Thinking that this should be after the loop as well?

Reply:
quite right! updated and moved